samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/ui/QrCodesScreen.kt

```diff
@@ -0,0 +1,89 @@
+package com.android.identity.testapp.ui
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.android.identity.appsupport.ui.qrcode.ShowQrCodeDialog
+import com.android.identity.appsupport.ui.qrcode.ScanQrCodeDialog
+
+@Composable
+fun QrCodesScreen(
+    showToast: (message: String) -> Unit
+) {
+    val showMdocQrCodeDialog = remember { mutableStateOf(false) }
+    val showUrlQrCodeDialog = remember { mutableStateOf(false) }
+    val showQrScanDialog = remember { mutableStateOf(false) }
+
+    if (showMdocQrCodeDialog.value) {
+        ShowQrCodeDialog(
+            title = "Scan code on reader",
+            description = "Your personal information won't be shared yet. You don't need to hand your phone to anyone to share your ID.",
+            dismissButton = "Close",
+            data = "mdoc:owBjMS4wAYIB2BhYS6QBAiABIVggWojRgrzl9C76WZQ/MzWdAuipaP8onZPl" +
+                    "+kRLYkNDFn/iJYILFujPhY3cdpBAe6YdTDOCNwqM/PPeaqZy/GClV6oy/GcCgYMCAaMA9AH1C1BF7+90KyxIN6kKOw4dBaaRBw==",
+            onDismiss = { showMdocQrCodeDialog.value = false }
+        )
+    }
+
+    if (showUrlQrCodeDialog.value) {
+        ShowQrCodeDialog(
+            title = "Scan code with phone",
+            description = "This is a QR code for https://github.com/openwallet-foundation-labs/identity-credential",
+            dismissButton = "Close",
+            data = "https://github.com/openwallet-foundation-labs/identity-credential",
+            onDismiss = { showUrlQrCodeDialog.value = false }
+        )
+    }
+
+    if (showQrScanDialog.value) {
+        ScanQrCodeDialog(
+            title = "Scan code",
+            description = "Ask the person you wish to request identity attributes from to present" +
+                    " a QR code. This is usually in their identity wallet.",
+            dismissButton = "Close",
+            onCodeScanned = { data ->
+                if (data.startsWith("mdoc:")) {
+                    showToast("Scanned mdoc URI $data")
+                    showQrScanDialog.value = false
+                    true
+                } else {
+                    false
```

Comment:
I know this is just a test app... ...but do we want to show a message somewhere that indicates that the scanned QR code doesn't appear to be an identity document?

Reply:
It just keeps scanning until a mdoc:// URI is encountered. I think that's working as intended.